lua/autorun/server/sv_daily_restart.lua

```diff
@@ -219,19 +409,65 @@ local function waitUntilRestartHour()
     createRestartTimer( secondsToWait )
 end
 
+local function waitForNextSoftRestartWindow()
+    local window = SOFT_RESTART_WINDOWS[currentSoftRestartWindow]
+
+    if not window then return end -- Only happens if none of the windows have a huge playerMax to guarantee being ran
+
+    local timeUntilNextWindowAlert = window.timeSinceStart * SECONDS_IN_HOUR - timeSinceStart() - LARGEST_ALERT_INTERVAL
+
+    if os.time() + timeUntilNextWindowAlert + RESTART_BUFFER < EARLIEST_RESTART_TIME then return end -- Hard restart is too close
+
+    timer.Create( SoftRestartTimerName, timeUntilNextWindowAlert, 1, function()
+        if #player.GetHumans() <= window.playerMax then
+            softRestartImminent = true
+            onSoftAlertTimeout()
+        else
+            currentSoftRestartWindow = currentSoftRestartWindow + 1
+            waitForNextSoftRestartWindow() 
+        end
+    end )
+end
+
 local function test_waitUntilRestartHour()
     EARLIEST_RESTART_TIME = os.time() + 100
     createRestartTimer( 0 )
 end
 
 initializeAlertIntervals()
 if TESTING_BOOLEAN then
+    LARGEST_ALERT_INTERVAL = TestAlertIntervalsInSeconds[1]
+
     test_waitUntilRestartHour()
 else
+    LARGEST_ALERT_INTERVAL = BaseAlertIntervalsInSeconds[1]
+
     waitUntilRestartHour()
+    waitForNextSoftRestartWindow()
 end
 
 
 DailyRestartTests.renew = function()
     test_waitUntilRestartHour()
 end
+
+hook.Add( "PlayerSay", "CFC_DailyRestart_StopSoftRestart", function( ply, msg )
+    if msg ~= SOFT_RESTART_STOP_COMMAND then return end
+    if not IsValid( ply ) then return end
+
+    if not canStopSoftRestart( ply ) then
+        ply:ChatPrint( "You do not have access to that command!" )
+        
+        return ""
+    end
+    
+    timer.Remove( SoftRestartTimerName )
+    initializeAlertIntervals()
+
+    if currentSoftRestartWindow < #SOFT_RESTART_WINDOWS then
+        currentSoftRestartWindow = currentSoftRestartWindow + 1
+        waitForNextSoftRestartWindow()
+    end
+
+    sendAlertToClients( "The soft restart has been canceled." )
+end )
```

Comment:
Maybe you could make this a ULX command? It'd probably be less code

Reply:
It'd make ULX a dependency, and it may even end up requiring _more_ code to make things integrate properly.

Reply:
Nah you could just make the ulx module, then decide if ulx doesn't exist to fall back to the regular method.

Exposing a function here to cancel the next restart, then a 6 line command definition to use that function.

I think it's very doable and lets us easily change who can use the command.